buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/PerformanceTest.java

```diff
@@ -79,44 +132,82 @@ public String getRuns() {
     @Option(option = "checks", description = "Tells which regressions to check. One of [none, speed, all]")
     public void setChecks(@Nullable String checks) {
         this.checks = checks;
-        systemProperty("org.gradle.performance.execution.checks", checks);
     }
 
     @Nullable @Optional @Input
     public String getChecks() {
         return checks;
     }
 
-    @Nullable @Optional @Input
-    public String getChannel() {
-        return channel;
-    }
-
     @Option(option = "channel", description = "Channel to use when running the performance test. By default, 'commits'.")
     public void setChannel(@Nullable String channel) {
         this.channel = channel;
-        systemProperty("org.gradle.performance.execution.channel", channel);
     }
 
-    @OutputDirectory
-    public File getDebugArtifactsDirectory() {
-        return debugArtifactsDirectory;
+    @Nullable @Optional @Input
+    public String getChannel() {
+        return channel;
     }
 
-    public void setDebugArtifactsDirectory(File debugArtifactsDirectory) {
-        this.debugArtifactsDirectory = debugArtifactsDirectory;
+    public void setResultStoreClass(String resultStoreClass) {
+        this.resultStoreClass = resultStoreClass;
+    }
+
+    @Nullable @Optional @Input
+    public String getResultStoreClass() {
+        return resultStoreClass;
     }
 
     @Option(option = "flamegraphs", description = "If set to 'true', activates flamegraphs and stores them into the 'flames' directory name under the debug artifacts directory.")
     public void setFlamegraphs(String flamegraphs) {
-        if ("true".equals(flamegraphs)) {
+        this.flamegraphs = "true".equals(flamegraphs);
+    }
+
+    @Nullable @Optional @Input
+    public Boolean isFlamegraphs() {
+        return flamegraphs;
+    }
+
+    @Input
+    public String getDatabaseUrl() {
```

Comment:
Is the databaseUrl an optional input? Or do we need to set it?

Reply:
I think it's not optional, all TC performance builds have this value in start parameters.

Reply:
I was wondering what happens if I run a performance test on my computer...

Reply:
Looks like when the URL is not set, then we use a local database: https://github.com/gradle/gradle/blob/0287f30f751644cac2a515e07c24a9e6d16ba030/subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/PerformanceDatabase.java#L64-L71

I guess we should use that Database for the report, too? And the URL should be optional or set to this default. WDYT?

Reply:
Since performance test is a very time-consuming operation, I doubt if this still makes sense on local machine. @oehme I'm curious about the original intention here?